src/network/networkd-address.c

```diff
@@ -511,6 +511,36 @@ static int address_remove_handler(sd_netlink *rtnl, sd_netlink_message *m, Link
         return 1;
 }
 
+static int address_set_netlink_message(const Address *address, sd_netlink_message *req, Link *link) {
+        uint32_t flags;
+        int r;
+
+        assert(address);
+        assert(req);
+        assert(link);
+
+        r = sd_rtnl_message_addr_set_prefixlen(req, address->prefixlen);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not set prefixlen: %m");
+
+        flags = address->flags | IFA_F_PERMANENT;
```

Comment:
While you are here refactoring, could you please add a couple of lines of comments explaining this flags manipulation (why is it necessary etc)? It's not immediately obvious without knowing the kernel inner working details in advance

Reply:
I noticed that IFA_F_PERMANENT flag is not necessary. The flag assignment is dropped.